File: Easy/1582. Special Positions in a Binary Matrix.java
class Solution {
    public int numSpecial(int[][] mat) {
        final int R = mat.length, C = mat[0].length;
        int[] flatRow = new int[R];
        int[] flatCol = new int[C];
        for (int r = 0; r < R; r++) {
            for (int c = 0; c < C; c++) {
                flatRow[r] += mat[r][c];
                flatCol[c] += mat[r][c];
            }
        }
        int numOfSpecial = 0;
        for (int r = 0; r < R; r++) {
            for (int c = 0; c < C; c++) {
                if (mat[r][c] == 1) {
                    if (flatRow[r] == 1 && flatCol[c] == 1) {
                        numOfSpecial += 1;
                    }
                }
            }
        }
        return numOfSpecial;
    }
}

File: Easy/1582. Special Positions in a Binary Matrix.py
from functools import cache
from typing import List


class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        @cache
        def colSum(col: int) -> int:
            return sum(row[col] for row in mat)
        res = 0
        for row in mat:
            if sum(row) == 1:
                col = row.index(1)
                if colSum(col) == 1:
                    res += 1
        return res


class Solution1:
    def numSpecial(self, mat: List[List[int]]) -> int:
        cols = [0] * len(mat[0])
        for row in mat:
            for col, num in enumerate(row):
                cols[col] += num
        res = 0
        for row in mat:
            if not (row.count(1) == 1):
                continue
            for col, num in enumerate(row):
                if num and cols[col] == 1:
                    res += 1
                    break
        return res